File: src/game/item/HealthPack.java
package game.item;

import game.TextureManager;
import game.Vector;
import game.creature.Creature;
import game.creature.Player;

public class HealthPack extends Item {
	public HealthPack(Vector pos) {
		super(pos, TextureManager.HEALTHPACK);
		color = new float[] {1, 0, 0};
	}
	
	@Override
	public boolean pickedUpBy(Creature c) {
		if (c instanceof Player) {
			c.health += 1;
			return true;
		} else {
			return false;
		}
	}
}

File: ts/game/item/HealthPack.ts
import { TextureManager } from '../TextureManager.js';
import { Vector } from '../Vector.js';
import { Creature } from '../creature/Creature.js';
import { Player } from '../creature/Player.js';
import { Item } from './Item.js';

export class HealthPack extends Item {
  constructor(pos: Vector) {
    super(pos, TextureManager.HEALTHPACK);
  }

  pickedUpBy(c: Creature): boolean {
    if (c instanceof Player) {
      c.health += 1;
      return true;
    }
    return false;
  }
}
